src/iframe-helper.js

```diff
@@ -132,8 +132,7 @@ function getListenForEvents(parentWin, sentinel, origin, triggerWin) {
     if (!contentWindow) {
       setTimeout(dropListenSentinel, 0, listenSentinel);
     } else if (sentinel === 'amp') {
-      // A non-3P code path, origin must match.
-      if (we.origin === origin && contentWindow == triggerWin) {
+      if (contentWindow == triggerWin) {
```

Comment:
Thanks @samjacoby!

@dvoytenko @lannka: When a message comes from `srcdoc`, the `origin` is `null` by design. Do you remember why we check origin of message in non-3P case in addition to contentWindow?  The 3P case never checks origin. I am hoping this is safe to remove. Thoughts? (Maybe a safer solution is checking whether origin is `null OR equal`)

Reply:
(this would also solve another bug where `resize` request was defined if `allow-same-origin` was not specified: https://github.com/ampproject/amphtml/issues/8819. Same issue, origin is `null` in those cases)

Reply:
In theory, a 3p iframe can redirect. srcdoc could too. I'd say, we should simply allow `origin == null` in our checks when we also control for `contentWindow`.

Reply:
So something like: `if ((we.origin === null || we.origin === origin) && contentWindow == triggerWin))`?

Also — I signed the CLA. Apologies for not doing so prior.

Reply:
More like `if (contentWindow == triggerWin && (we.origin == origin || we.origin == null && itsASrcDocIframe))`. I.e. a bit more strict on a side where we.origin is null.

Reply:
'itsASrcDocIframe' check is too restrict IMO. we have the same issue when `src` is used but `allow-same-origin` is not set in which case browsers also set `origin` to null

Reply:
Maybe. I'm a bit concerned about opening this hole a little too wide. Really not a big fun of how `allow-same-origin` is defined.

Reply:
Discussed offline: New proposal is:

- add a new `opt_isSnsensitive // If set, data is not considered sensitive and origin check does not happen to listenFor()`
- Completely ignore origin check if set. This also fixes issues where iframe may have done a legit redirect as those cases also are refused resize requests.
- Have `amp-iframe` set the `opt_isSnsensitive` to `true` to `resize` event.

@samjacoby This is becoming a bit bigger than expected, up to you if you like to continue with the changes or pass it to us. Either way, we really appreciate your involvement.

Reply:
Happy to continue to move it along, if you've got the time to hold my hand and it won't slow things down. I think I'll need fairly specific pointers, as I'm not deeply familiar with the underlying security concerns, and don't want to slip up.

> - add a new opt_isSnsensitive // If set, data is not considered sensitive and origin check does not happen to listenFor()

This would be adding it as an argument here, right: https://github.com/ampproject/amphtml/blob/master/src/iframe-helper.js#L250
> - Completely ignore origin check if set. This also fixes issues where iframe may have done a legit redirect as those cases also are refused resize requests.

Something like...`if (contentWindow == triggerWin && (we.origin == origin || opt_isSnsensitive === true)`?
> - Have amp-iframe set the `opt_isSnsensitive` to true to resize event.

And set the `opt_isSnsensitive`  to be `true` here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/0.1/amp-iframe.js#L397

If it'd be more expedient for one of you to hammer it out, that's fine. My main priority is just to see a fix in production, though I would like to contribute!

Reply:
all correct (except my misspelling :) `opt_isInsensitive`)
+ updating/adding some tests

Let's plan this way: if you find time over the next week or so, go for it. Otherwise I can take over in about 2 weeks when my time frees up.

Reply:
> all correct (except my misspelling :) opt_isInsensitive)

Ha — I was wondering about that. Figured it was something. 😀 

I took a pass at adding the `opt_isInsensitive`, but I'm not sure I hooked it through in the right way — I added an additional parameter to the `windowEvents` object, that's checked when the listener is fired, but that seems like it may permit other types of events to be executed w/o an origin check as well. At any rate, throwing it out there for now! Thanks for taking a look.